lib/src/entity.dart

```diff
@@ -0,0 +1,30 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'realm_class.dart';
+
+mixin RealmEntity {
+  Realm? _realm;
+  Realm get realm => _realm ?? (throw RealmException('$this not managed'));
+  bool get isManaged => _realm != null;
+}
+
+extension RealmEntityInternal on RealmEntity {
+  void setRealm(Realm value) => _realm = value;
```

Comment:
As I suggest in a comment I suggest we move RealmEntity into `realm_object.dart`. Then we avoid the need to expose this method on all user types.

Reply:
It is needed in `list.dart` as well. I use the common pattern of create a public class that is hidden outside package.